lock/src/main/java/org/infinispan/lock/impl/functions/UnlockFunction.java

```diff
@@ -32,29 +32,46 @@
  * @author Katia Aresti, karesti@redhat.com
  * @since 9.2
  */
+@ProtoTypeId(ProtoStreamTypeIds.CLUSTERED_LOCK_FUNCTION_UNLOCK)
 public class UnlockFunction implements Function<EntryView.ReadWriteEntryView<ClusteredLockKey, ClusteredLockValue>, Boolean> {
 
    private static final Log log = LogFactory.getLog(UnlockFunction.class, Log.class);
 
-   public static final AdvancedExternalizer<UnlockFunction> EXTERNALIZER = new Externalizer();
-
    private final String requestId;
-   private final Set<Object> requestors;
+   private final Set<Object> requesters;
 
    public UnlockFunction(Object requestor) {
       this.requestId = null;
-      this.requestors = Collections.singleton(requestor);
+      this.requesters = Collections.singleton(requestor);
    }
 
-   public UnlockFunction(String requestId, Set<Object> requestors) {
+   public UnlockFunction(String requestId, Set<Object> requesters) {
       this.requestId = requestId;
-      this.requestors = requestors;
+      this.requesters = requesters;
+   }
+
+   @ProtoFactory
+   static UnlockFunction protoFactory(String requestId, Set<WrappedMessage> requesters) {
+      return new UnlockFunction(
+            requestId,
+            requesters == null ? null : requesters.stream().map(WrappedMessage::getValue).collect(Collectors.toSet())
+      );
+   }
+
+   @ProtoField(1)
+   String getRequestId() {
+      return requestId;
+   }
+
+   @ProtoField(2)
+   Set<WrappedMessage> getRequesters() {
```

Comment:
set of addresses :crossed_fingers:

Reply:
The best we can do is a Stream of `JGroupAddress` as Protostream does not know how to handle `Set<Address> getRequesters()` as the type is abstract.